config/migrate.ts

```diff
@@ -0,0 +1,209 @@
+  deleteConfigFile as newDeleteConfigFile,
+} from './configFile';
+import {
+  GLOBAL_CONFIG_PATH,
+  DEFAULT_CMS_PUBLISH_MODE,
+  HTTP_TIMEOUT,
+  ENV,
+  HTTP_USE_LOCALHOST,
+  DEFAULT_ACCOUNT,
+  DEFAULT_PORTAL,
+} from '../constants/config';
+import { i18n } from '../utils/lang';
+
+const i18nKey = 'config.migrate';
+
+export function getConfig(
+  useHiddenConfig?: boolean,
+  configPath?: string
+): Partial<CLIConfig> | null {
+  if (useHiddenConfig) {
+    return CLIConfiguration.config;
+  }
+  return config_DEPRECATED.loadConfig(configPath);
+}
+
+export function getConfigPath(
+  configPath?: string,
+  useHiddenConfig = false
+): string | null {
+  if (useHiddenConfig) {
+    return getConfigFilePath();
+  }
+  return config_DEPRECATED.getConfigPath(configPath);
+}
+
+export function configFileExists(
+  useHiddenConfig?: boolean,
+  configPath?: string
+): boolean {
+  return useHiddenConfig
+    ? newConfigFileExists()
+    : Boolean(config_DEPRECATED.getConfigPath(configPath));
+}
+
+export function deleteConfigFile(useHiddenConfig = false): void {
+  if (useHiddenConfig) {
+    newDeleteConfigFile();
+  }
+  config_DEPRECATED.deleteConfigFile();
+}
+
+function writeGlobalConfigFile(updatedConfig: CLIConfig_NEW): void {
+  const updatedConfigJson = JSON.stringify(updatedConfig);
+  createEmptyConfigFile({}, true);
+  loadConfig('');
+
+  try {
+    writeConfig({ source: updatedConfigJson });
+    deleteConfigFile();
+  } catch (error) {
+    deleteEmptyConfigFile();
+    throw new Error(
+      i18n(`${i18nKey}.errors.writeConfig`, { configPath: GLOBAL_CONFIG_PATH }),
+      { cause: error }
+    );
+  }
+}
+
+export function migrateConfig(
+  deprecatedConfig: CLIConfig_DEPRECATED | null
+): void {
+  if (!deprecatedConfig) {
+    throw new Error(i18n(`${i18nKey}.errors.noDeprecatedConfig`));
+  }
+  const { defaultPortal, portals, ...rest } = deprecatedConfig;
+  const updatedConfig = {
+    ...rest,
+    defaultAccount: defaultPortal,
+    accounts: portals
+      .filter(({ portalId }) => portalId !== undefined)
+      .map(({ portalId, ...rest }) => ({
+        ...rest,
+        accountId: portalId!,
+      })),
+  };
+  writeGlobalConfigFile(updatedConfig);
+}
+
+type ConflictValue = boolean | string | number | CmsPublishMode | Environment;
+export type ConflictProperty = {
+  property: keyof CLIConfig_NEW;
+  oldValue: ConflictValue;
+  newValue: ConflictValue;
+};
+
+export function mergeConfigProperties(
+  globalConfig: CLIConfig_NEW,
+  deprecatedConfig: CLIConfig_DEPRECATED,
+  force?: boolean
+): {
+  initialConfig: CLIConfig_NEW;
+  conflicts: Array<ConflictProperty>;
+} {
+  const propertiesToCheck: Array<keyof Partial<CLIConfig>> = [
+    DEFAULT_CMS_PUBLISH_MODE,
+    HTTP_TIMEOUT,
+    ENV,
+    HTTP_USE_LOCALHOST,
+  ];
+  const conflicts: Array<ConflictProperty> = [];
+
+  propertiesToCheck.forEach(prop => {
+    if (prop in globalConfig && prop in deprecatedConfig) {
+      if (force) {
```

Comment:
Not a blocker, you could also DRY this up with `if (force || globalConfig[prop] == deprecatedConfig[prop])`